### PR TITLE
Plumb through diagnostics to show error bar for bad network quality in call

### PR DESCRIPTION
**Pull request by prprabhu-ms**
# What
* Add stateful plumbing for call diagnostics
* Add bindings to bubble up error when speaking while muted

KI: Currently, the error bar shown when speaking while muted can't be dimissed at all. A fix is pending a small rearchitecturing of ErrorBar.

# How Tested
storybook

### CODE REVIEW

**Comment by prprabhu-ms** [comment]
I'm still working on fixing the problem that the new error can't be dismissed. Might be worth landing this with the actual binding disabled (so all plumbing is in place) if fixing that issue takes longer.

PTAL.

File: packages/calling-component-bindings/src/errorBarSelector.ts [hunk 1]
```diff
@@ -15,40 +15,48 @@ import { createSelector } from 'reselect';
...
-  }
-  if (latestErrors['Call.mute'] !== undefined) {
-    activeErrors.push('muteGeneric');
-  }
-  if (latestErrors['Call.stopScreenSharing'] !== undefined) {
-    activeErrors.push('stopScreenShareGeneric');
-  }
+    // Errors reported via diagnostics are more reliable than from API method failures, so process those first.
+    if (diagnostics?.media.latest.speakingWhileMicrophoneIsMuted?.value) {
+      activeErrors.push('speakingWhileMuted');
```
Comment (prprabhu-ms) [inline packages/calling-component-bindings/src/errorBarSelector.ts:32]: Will comment out this line if I can't fix the KI first.

File: packages/calling-stateful-client/src/DiagnosticsSubscriber.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      }
+      const media = call.diagnostics?.media.latest;
+      if (media) {
+        media[args.diagnostic] = latestFromEvent(args);
+      }
+    });
+    this.logDiagnostics('media update');
+  }
+
+  private logDiagnostics(operation: string) {
```
Comment (prprabhu-ms) [inline packages/calling-stateful-client/src/DiagnosticsSubscriber.ts:86]: Will remove this before landing.

**Review by anjulgarg — APPROVED** [review]

**Comment by anjulgarg** [comment]
LGTM
Not sure if showing an error bar is the best way to go for indicating to a user that they are speaking while their mic is off.

**Review by alcail — APPROVED** [review]

**Review by alkwa-msft — APPROVED** [review]

File: packages/calling-component-bindings/src/errorBarSelector.ts (hunk 1 above)
Comment (prprabhu-ms) [reply packages/calling-component-bindings/src/errorBarSelector.ts:32]: KI fixed in main.

**Comment by prprabhu-ms** [comment]
> LGTM
> Not sure if showing an error bar is the best way to go for indicating to a user that they are speaking while their mic is off.

Teams shows it as a similar banner, though we may not want it to be quite the error banner.

Anyway, I had picked an easy one just to showcase the plumbing. Switched to surfacing network connection error -- the thing I actually was working towards.

**Pull request merged**